api/envoy/config/route/v3/route_components.proto

```diff
@@ -229,11 +229,18 @@ message Route {
     DirectResponseAction direct_response = 7;
 
     // [#not-implemented-hide:]
-    // If true, a filter will define the action (e.g., it could dynamically generate the
+    // A filter-defined action (e.g., it could dynamically generate the
     // RouteAction).
     // [#comment: TODO(samflattery): Remove cleanup in route_fuzz_test.cc when
     // implemented]
     FilterAction filter_action = 17;
+
+    // [#not-implemented-hide:]
+    // An action used when the route will generate a response directly,
+    // without forwarding to an upstream host. This will be used in non-proxy
+    // xDS clients like the gRPC server. It could also be used in the future
+    // in Envoy for a filter that directly generates responses for requests.
+    NonForwardingAction non_forwarding_action = 18;
```

Comment:
This makes sense. Just thinking loudly, and probably not worth the churn, shall other 'non-forwarding' actions (e.g. direct response and redirect) to be included here in the future?

Reply:
I think they'll probably remain separate.  The main difference between those existing action types and this one is that those existing action types explicitly define everything that's needed to process the request (e.g., "reply with this specific HTTP status" or "redirect to this specific URL"), whereas this new type of action is basically saying "we won't be forwarding the request, but we're leaving it undefined how it will actually be handled".  Some external code (e.g., the gRPC server application or an Envoy filter) will have to actually decide how to respond to the request.